packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3536,11 +3553,71 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  final bool _platformSupportsFadeOnScroll = defaultTargetPlatform == TargetPlatform.android
+                                          || defaultTargetPlatform == TargetPlatform.iOS;
+
   void _onEditableScroll() {
-    _selectionOverlay?.updateForScroll();
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      _selectionOverlay?.updateForScroll();
+    }
     _scribbleCacheKey = null;
   }
 
+  void _handleContextMenuOnParentScroll(ScrollNotification notification) {
+    if (_internalScrolling) {
+      return;
+    }
+    _handleContextMenuOnScroll(notification);
+  }
+
+  void _handleContextMenuOnScroll(ScrollNotification notification) {
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      return;
+    }
+    // When the scroll begins and the toolbar is visible, hide it
+    // until scrolling ends.
+    //
+    // When scrolling ends and the toolbar was present at the
+    // beginning of the scroll we should re-show the toolbar only
+    // if the selection endpoints are within the viewport at the
+    // end of the scroll. If not then we should schedule the
+    // toolbar to be shown when we scroll the selection back into
+    // view.
+    //
+    // A scheduled toolbar will only be shown if the editing state
+    // did not change between the time the toolbar was first
+    // scheduled to be shown to when it is ready to be shown.
+    if (notification is ScrollStartNotification) {
+      _toolbarVisibleAtScrollStart = _selectionOverlay != null && _selectionOverlay!.toolbarIsVisible;
+      if (_toolbarVisibleAtScrollStart) {
+        hideToolbar(false);
+      }
+    } else if (notification is ScrollEndNotification) {
+      final Rect paintBounds = renderEditable.paintBounds;
```

Comment:
Does anyone know if `renderEditable.getTransformTo(null) == Matrix4.zero()` is an accurate way of telling if a render object is in the current viewport? From the documentation of `applyPaintTransform` which is called during `getTransformTo`, this sounds like it is accurate

```dart
Some RenderObjects will provide a zeroed out matrix in this method, indicating that the child should not paint anything or respond to hit tests currently. A parent may supply a non-zero matrix even though it does not paint its child currently, for example if the parent is a [RenderOffstage] with offstage set to true. In both of these cases, the parent must return false from [paintsChild].
```

and from `defaultApplyPaintTransform` which is called from `applyPaintTransform`.

```dart
  /// Applies the transform that would be applied when painting the given child
  /// to the given matrix.
  ///
  /// Render children whose [TextParentData.offset] is null zeros out the
  /// `transform` to indicate they're invisible thus should not be painted.
```